ejs/index.d.ts

```diff
@@ -11,7 +11,12 @@ declare namespace Ejs {
     function resolveInclude(name: string, filename: string): string;
     function compile(template: string, opts?: Options): (TemplateFunction);
     function render(template: string, data?: Data, opts?: Options): string;
-    function renderFile(path: string, data?: Data, opts?: Options, cb?: Function): any;// TODO RenderFileCallback return type
+
+    type RenderFileCallback<T> = (err: Error, str?: string) => T;
+    function renderFile<T>(path: string, cb: RenderFileCallback<T>): T;
+    function renderFile<T>(path: string, data: Data, cb: RenderFileCallback<T>): T;
+    function renderFile<T>(path: string, data: Data, opts: Options, cb: RenderFileCallback<T>): T;
+
```

Comment:
@anubmat I don't know TS that well, but as far as I know you can't put required parameters after optional ones? The reason I thought multiple overloads are needed is that the API has required arguments as both the first and last parameters.